### DISCUSSION TITLE
Is it possible to detect console's default background color (dark vs light)?

### DISCUSSION THREAD

**Discussion by plannigan** [Textualize/rich#2807] [Q&A] [answered]
I saw in [this question](https://github.com/Textualize/rich/discussions/990) that it is possible to say "use the default background color". However, I haven't been able to find a way to ask "what is the default background color?". If possible, I could dynamically pick which theme to use so that id doesn't clash with what the user is currently using for their terminal.

**Comment by willmcgugan** [comment, marked as answer]
It's possible on some terminals, but not all. This is outside of the scope of Rich, but you could implement the method that gets the current theme on supported terminals.

**Reply by plannigan** [reply]
That makes sense. Based on your comment I found this [previous question](https://github.com/Textualize/rich/discussions/1170), which has a bit more detail. (I guess I'm bad a searching).
